kafka/src/main/java/io/micronaut/configuration/kafka/KafkaProducerFactory.java

```diff
@@ -130,6 +124,21 @@ public <K, V> Producer<K, V> getProducer(
         return getKafkaProducer(id, null, k, v, false);
     }
 
+    /**
+     *
+     * Creates kafka producer, could be overridden for further control.
+     *
+     * @param config properties for producer
+     * @param ks key serializer
+     * @param vs value serializer
+     * @param <K> key type
+     * @param <V> value type
+     * @return new instance of producer
+     */
+    protected <K, V> Producer<K, V> createProducer(Properties config, Serializer<K> ks, Serializer<V> vs) {
```

Comment:
Add nullability annotations and `@since` declaration

Reply:
Added.